src/request.js

```diff
@@ -146,19 +146,47 @@ class Request {
     /**
      * Stores information about an error that occurred during processing of this request.
      *
+     * You should always use Error instances when throwing errors in JavaScript.
+     *
+     * Nevertheless, to improve the debugging experience when using third party libraries
+     * that may not always throw an Error instance, the function performs a type
+     * inspection of the passed argument and attempts to extract as much information
+     * as possible, since just throwing a bad type error makes any debugging rather difficult.
+     *
      * @param {Error|String} errorOrMessage Error object or error message to be stored in the request.
+     * @param {Object} [options]
+     * @param {Boolean} [options.omitStack=false] Only push the error message without stack trace when true.
      */
-    pushErrorMessage(errorOrMessage) {
-        if (!_.isString(errorOrMessage) && !(errorOrMessage instanceof Error)) {
-            throw new Error('Parameter errorOrMessage must be a String or an instance of Error');
+    pushErrorMessage(errorOrMessage, options = {}) {
+        const { omitStack } = options;
+        let message;
+        const type = typeof errorOrMessage;
+        if (type === 'object') {
+            if (!errorOrMessage) {
+                message = 'null';
+            } else if (errorOrMessage instanceof Error) {
+                message = omitStack
+                    ? errorOrMessage.message
+                    // Use inspect because .toString() just returns the message.
+                    : util.inspect(errorOrMessage);
+            } else if (errorOrMessage.message) {
+                message = errorOrMessage.message; // eslint-disable-line prefer-destructuring
+            } else if (errorOrMessage.toString() !== '[object Object]') {
+                message = errorOrMessage.toString();
+            } else {
+                try {
+                    message = util.inspect(errorOrMessage);
+                } catch (err) {
+                    message = 'Unable to extract any message from the received object.';
```

Comment:
Can this ever happen? Isn't util.inspect() bulletproof?

Reply:
BTW we use it before without try {} catch {}, I'd just make it all same and get rid of try-catch.